lib/waterline/query/finders/basic.js

```diff
@@ -45,6 +46,10 @@ module.exports = {
     // Normalize criteria
     criteria = normalize.criteria(criteria);
 
+    this.beforeCallbacks.call(self,criteria.where,function(err){
```

Comment:
So, this guys first point is a problem I think. If no criteria is passed to `Model.find` then `values` will be null when it gets into the `beforeFind` callback. 

https://github.com/balderdashy/waterline/pull/902#issuecomment-110136774

Might need to do `criteria.where || {}` or something like that...not totally sure of the implications there.